core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -159,7 +164,10 @@ class NewHadoopRDD[K, V](
 
       private def close() {
         try {
-          reader.close()
+          if (!finished) {
```

Comment:
As @srowen mentioned previously, this won't close the `reader` if calling `close` before reaching at the end of stream. E.g., if there are 10 items in this Iterator, but the user only uses the first item and then calls `close`.